### PR TITLE
Simplify Program printing code to improve debugging efficiency

### PR DESCRIPTION
**Pull request by chenwhql**
In order to get simpler and clearer debug strings, add `_to_readable_code` method to `Program`, `Block`, `Operator`, `Variable`,  and change method `__str__` to call `to_readable_code` method by default.

- There is almost no information loss in this conversion, but the information density is increased

- why add  new API instead of changing old method `to_string`?
  - `to_string` interface has been existed for a long time, not sure if changing it will cause compatibility problems for some users
  - keep `to_string` interface unchanged, users still can easily get the original ptotobuf string through `to_string`, or `print(Program.to_string())`

----

For a same program:

- The new debug string example:
```
{ // block 0
    var X : fluid.VarType.LOD_TENSOR.shape(-1, 23, 48).astype(VarType.FP32)
    persist trainable param W : fluid.VarType.LOD_TENSOR.shape(23, 48).astype(VarType.FP32)

    {Out=['X']} = abs(inputs={X=['X']}, op_device = , op_namescope = /, op_role = 0, op_role_var = [], use_cudnn = False, use_mkldnn = False)
}
```
- The old debug string example:
```
blocks {
  idx: 0
  parent_idx: -1
  vars {
    name: "W"
    type {
      type: LOD_TENSOR
      lod_tensor {
        tensor {
          data_type: FP32
          dims: 23
          dims: 48
        }
      }
    }
    persistable: true
  }
  vars {
    name: "X"
    type {
      type: LOD_TENSOR
      lod_tensor {
        tensor {
          data_type: FP32
          dims: -1
          dims: 23
          dims: 48
        }
        lod_level: 0
      }
    }
  }
  ops {
    inputs {
      parameter: "X"
      arguments: "X"
    }
    outputs {
      parameter: "Out"
      arguments: "X"
    }
    type: "abs"
    attrs {
      name: "op_device"
      type: STRING
      s: ""
    }
    attrs {
      name: "use_cudnn"
      type: BOOLEAN
      b: false
    }
    attrs {
      name: "op_role"
      type: INT
      i: 0
    }
    attrs {
      name: "op_role_var"
      type: STRINGS
    }
    attrs {
      name: "op_callstack"
      type: STRINGS
      strings: "  File \"/usr/local/lib/python3.5/dist-packages/paddle/fluid/framework.py\", line 2730, in append_op\n    attrs=kwargs.get(\"attrs\", None))\n"
      strings: "  File \"../python/paddle/fluid/tests/unittests/test_program_code.py\", line 90, in setUp\n    type=\"abs\", inputs={\"X\": [self.var]}, outputs={\"Out\": [self.var]})\n"
      strings: "  File \"/usr/lib/python3.5/unittest/case.py\", line 596, in run\n    self.setUp()\n"
      strings: "  File \"/usr/lib/python3.5/unittest/case.py\", line 648, in __call__\n    return self.run(*args, **kwds)\n"
      strings: "  File \"/usr/lib/python3.5/unittest/suite.py\", line 122, in run\n    test(result)\n"
      strings: "  File \"/usr/lib/python3.5/unittest/suite.py\", line 84, in __call__\n    return self.run(*args, **kwds)\n"
      strings: "  File \"/usr/lib/python3.5/unittest/suite.py\", line 122, in run\n    test(result)\n"
      strings: "  File \"/usr/lib/python3.5/unittest/suite.py\", line 84, in __call__\n    return self.run(*args, **kwds)\n"
      strings: "  File \"/usr/lib/python3.5/unittest/runner.py\", line 176, in run\n    test(result)\n"
      strings: "  File \"/usr/lib/python3.5/unittest/main.py\", line 255, in runTests\n    self.result = testRunner.run(self.test)\n"
      strings: "  File \"/usr/lib/python3.5/unittest/main.py\", line 94, in __init__\n    self.runTests()\n"
      strings: "  File \"../python/paddle/fluid/tests/unittests/test_program_code.py\", line 117, in <module>\n    unittest.main()\n"
    }
    attrs {
      name: "op_namescope"
      type: STRING
      s: "/"
    }
    attrs {
      name: "use_mkldnn"
      type: BOOLEAN
      b: false
    }
  }
}
version {
  version: 0
}
```

### CODE REVIEW

**Review by gongweibao — CHANGES_REQUESTED** [review]
我之前的那个文件删除吧！

**Comment by chenwhql** [comment]
> 我之前的那个文件删除吧！

done，还有几个工具函数在用，只删除了重复的

**Comment by gongweibao** [comment]
单测没用的话可以删除。^_^

**Comment by chenwhql** [comment]
> 单测没用的话可以删除。^_^

好的，已删除，^_^

File: python/paddle/fluid/tests/unittests/test_program_code.py [hunk 1]
```diff
@@ -27,55 +27,46 @@
...
-        place = fluid.CPUPlace()
-        self.init_serv(place)
-        self.init_client(place, 9123)
+    def append_cond_op(self, program):
+        def true_func():
+            return layers.fill_constant(shape=[2, 3], dtype='int32', value=2)
 
-    def init_serv(self, place):
-        main = fluid.Program()
+        def false_func():
```
Comment (gongweibao) [inline python/paddle/fluid/tests/unittests/test_program_code.py:96]: 这个文件只需要改动`program_to_code(main)`这一行吧？为何把组网也改了？

Comment (chenwhql) [reply python/paddle/fluid/tests/unittests/test_program_code.py:96]: 上一个commit就是只改了这一行，然后又提了一次把旧单测整个移除了，下面新增了单测，git识别成这样了，其实是分开的

Comment (chenwhql) [reply python/paddle/fluid/tests/unittests/test_program_code.py:48]: reset到上个commit就是只改一行，要退回去吗

Comment (chenwhql) [reply python/paddle/fluid/tests/unittests/test_program_code.py:96]: 恢复了删除的单测，保留也没什么影响，下面新增的单测是为了通过覆盖率

**Review by gongweibao — APPROVED** [review]
LGTM

**Review by luotao1 — APPROVED** [review]

**Review by lanxianghit — APPROVED** [review]

**Review by phlrain — APPROVED** [review]

**Pull request merged**